packages/core/data-transfer/lib/providers/local-strapi-source-provider/entities.ts

```diff
@@ -58,10 +59,58 @@ export const createEntitiesTransformStream = (): PassThrough => {
 /**
  * Get the list of attributes that needs to be populated for the entities streaming
  */
-const getPopulateAttributes = (contentType: ContentTypeSchema) => {
+const getPopulateAttributes = (strapi: Strapi.Strapi, contentType: ContentTypeSchema) => {
   const { attributes } = contentType;
 
-  return Object.keys(attributes).filter((key) =>
-    ['component', 'dynamiczone'].includes(attributes[key].type)
-  );
+  const populate: any = {};
+
+  const entries: [string, any][] = Object.entries(attributes);
+
+  for (const [key, attribute] of entries) {
+    if (attribute.type === 'component') {
+      const component = strapi.getModel(attribute.component);
+      const subPopulate = getPopulateAttributes(strapi, component);
```

Comment:
Is it possible for a component or DZ to reference itself? Even if it's not technically allowed but theoretically possible, we should check here to prevent infinite recursion.

Reply:
Indeed,

Two possible solutions:
- Having a max-depth option to limit the circular dependencies (that we could expose through some sort of config)
- Not exporting the circular dependencies
- A combination of both where we populate circular dependencies up to a certain amount of time

What do you think?

Also ping @alexandrebodin on this one

Reply:
The more common case is when components have relations to content-types that endup retargeting the same component. The schema can be circular even if the data isn't. 

I would recommend to build a populate tree go as deep as you can until you encrounter the known uid from your current branch in the tree

Reply:
We're not populating relations anyway here. Only components & dynamic zones (at nested levels)

Reply:
>until you encrounter the known uid from your current branch in the tree

Which would mean restricting circular dependencies between components & not exporting their data

Reply:
- I don't see a usecase where you would do a circular dependency inside your components. And you can't do it in the UI. 
- If you are advanced enough to create not just nested components with depth > 2 but circular dependencies you probably already migrate on your own. 

If this situation occurs we can always add a depth config but I would avoid it as long as possible as it is outside of the intended use